IoTHubBuddy/IoTViewer/MessageManager.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Devices.Geolocation;
+using Windows.UI.Xaml.Controls.Maps;
+
+namespace IoTHubBuddy
+{
+    class MessageManager
+    {
+        MapPage m_view;
+        public MessageManager(MapPage view)
+        {
+            m_view = view;
+        }
+        /// <summary>
+        /// parse message, check validity of message format, display coordinates or error message accordingly
+        /// </summary>
+        /// <param name="msg"></param>
+        public void parseMessage(string msg)
+        {
+            JObject jsonMsg = JObject.Parse(msg);
+            Debug.WriteLine(jsonMsg);
+            if (validateMessage(jsonMsg))
```

Comment:
I suggest doing validation in place to avoid extra lookups ( `var a = x["y"]; if(!a) a["z"] ...` )